carbon-mb/components/andes-rest-service/src/main/java/org/wso2/carbon/andes/service/exceptions/mappers/InternalServerErrorMapper.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.andes.service.exceptions.mappers;
+
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.osgi.service.component.annotations.Component;
+import org.wso2.carbon.andes.service.exceptions.InternalServerException;
+import org.wso2.carbon.andes.service.types.ErrorResponse;
+
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+
+/**
+ * Mapper class for {@link InternalServerException}.
+ */
+@Component(
+        name = "org.wso2.carbon.andes.exception.InternalServerErrorMapper",
+        service = ExceptionMapper.class,
+        immediate = true
+)
+public class InternalServerErrorMapper implements ExceptionMapper<InternalServerException> {
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Response toResponse(InternalServerException e) {
+        ErrorResponse errorResponse = new ErrorResponse();
+        errorResponse.setTitle("Error occurred in the server while performing an operation.");
+        errorResponse.setCode(500);
```

Comment:
http://docs.oracle.com/javaee/6/api/javax/ws/rs/core/Response.Status.html#INTERNAL_SERVER_ERROR shouldn't we be using this?